src/battle.ts

```diff
@@ -221,155 +222,171 @@ export default class Battle {
-            return "battle_legendary_gro_kyo";
-          }
-          if (pokemon.species.speciesId === Species.RAYQUAZA) {
-            return "battle_legendary_rayquaza";
-          }
-          if (pokemon.species.speciesId === Species.DEOXYS) {
-            return "battle_legendary_deoxys";
-          }
-          if (pokemon.species.speciesId === Species.UXIE || pokemon.species.speciesId === Species.MESPRIT || pokemon.species.speciesId === Species.AZELF) {
-            return "battle_legendary_lake_trio";
-          }
-          if (pokemon.species.speciesId === Species.HEATRAN || pokemon.species.speciesId === Species.CRESSELIA || pokemon.species.speciesId === Species.DARKRAI || pokemon.species.speciesId === Species.SHAYMIN) {
-            return "battle_legendary_sinnoh";
-          }
-          if (pokemon.species.speciesId === Species.DIALGA || pokemon.species.speciesId === Species.PALKIA) {
-            if (pokemon.getFormKey() === "") {
+        if (scene.musicPreference === MusicPreference.CONSISTENT) {
+          switch (pokemon.species.speciesId) {
+            case Species.REGIROCK:
+            case Species.REGICE:
+            case Species.REGISTEEL:
+            case Species.REGIGIGAS:
+            case Species.REGIDRAGO:
+            case Species.REGIELEKI:
+              return "battle_legendary_regis_g5";
+            case Species.KYUREM:
+              return "battle_legendary_kyurem";
+            default:
+              if (pokemon.species.legendary) {
+                return "battle_legendary_res_zek";
+              }
+              return "battle_legendary_unova";
+          }
+        } else if (scene.musicPreference === MusicPreference.MIXED) {
+          switch (pokemon.species.speciesId) {
+            case Species.ARTICUNO:
+            case Species.ZAPDOS:
+            case Species.MOLTRES:
+            case Species.MEWTWO:
+            case Species.MEW:
+              return "battle_legendary_kanto";
+            case Species.RAIKOU:
+              return "battle_legendary_raikou";
+            case Species.ENTEI:
+              return "battle_legendary_entei";
+            case Species.SUICUNE:
+              return "battle_legendary_suicune";
+            case Species.LUGIA:
+              return "battle_legendary_lugia";
+            case Species.HO_OH:
+              return "battle_legendary_ho_oh";
+            case Species.REGIROCK:
+            case Species.REGICE:
+            case Species.REGISTEEL:
+            case Species.REGIGIGAS:
+            case Species.REGIDRAGO:
+            case Species.REGIELEKI:
+              return "battle_legendary_regis_g6";
+            case Species.GROUDON:
+            case Species.KYOGRE:
+              return "battle_legendary_gro_kyo";
+            case Species.RAYQUAZA:
+              return "battle_legendary_rayquaza";
+            case Species.DEOXYS:
+              return "battle_legendary_deoxys";
+            case Species.HEATRAN:
```

Comment:
```suggestion
            case Species.UXIE:
            case Species.MESPRIT:
            case Species.AZELF:
              return "battle_legendary_lake_trio";
            case Species.HEATRAN:
```
I think you accidentally skipped these